antiweb.py

```diff
@@ -1465,18 +1465,156 @@ def filter_output(self, lines):
+    #@include(CSharpReader.filter_output doc)
+    #@(CSharpReader doc)
+    
+    def _accept_token(self, token):
+        return super(CSharpReader, self)._accept_token(token)
+    
+    def _cut_comment(self, index, token, text):
+        return super(CSharpReader, self)._cut_comment(index, token, text)
+        
+    #@cstart(CSharpReader.strip_tags)
+    def strip_tags(self,  soup):
+        """
+        .. py:method:: strip_tags(soup)
+
+           Removes all C# XML tags. The tags are replaced by their attributes and contents.
+
+           :param soup: The parsed xml tags.
+           :return: the XML tags replaced by their attributes and contents.
+        """
+        if isinstance(soup,  Tag) and not soup.contents:            
+            return self.get_attribute_text(soup)
+    
+        for tag in soup.find_all(True):
+            text = self.get_attribute_text(tag)
+            
+            #if the content is a Navigablestring the content is simply concatenated 
+            #otherwise the contents of the content are recursively checked for xml tags
+            for content in tag.contents:
+                if not isinstance(content, NavigableString):
+                    content = self.strip_tags(content)
+                text += content
+    
+            tag.replaceWith(text)
+    
+        return soup
+    #@edoc
+    
+    #@cstart(CSharpReader.get_attribute_text)
+    def get_attribute_text(self,  tag):
+        """
+        .. py:method:: get_attribute_text(tag)
+
+           Gets the values of all attributes of a XML tag.
+
+           :param tag: A BeautifulSoup Tag.
+           :return: the values of all attributes concatenated
+        """
+        #collect all values of xml tag attributes
+        attributes = tag.attrs
+        attribute_text = ""
+        for attribute, value in attributes.items():
+            attribute_text = value + " "
+        return attribute_text
+    #@edoc
+    
+    #@cstart(CSharpReader.filter_output)
+    def filter_output(self, lines):
+        """
+        .. py:method:: filter_output(lines)
+        
+           Applies C# specific filtering for the final output.
+           XML comment tags are replaced by their attributes and contents.
+
+           See :py:meth:`Reader.filter_output`
+           
+           :param lines: All lines of a file. The directives have already been replaced.
+           :return: A generator containing all lines for the final output.
+        """
+        #first the CReader filters the output
+        #afterwards the CSharpReader does some C# specific filtering
+        lines = super(CSharpReader, self).filter_output(lines)
+                
+        xml_lines_block = []
+        xml_start_index = -1
+        
+        for l in lines:
+            if l.type == "d":
+                #remove comment chars in document lines
+                stext = l.text.lstrip()
+
+                if stext == "/*" or stext == "*/":
```

Comment:
Is there a way we could factor out comment character sequences as constants, to be used by all Readers that need them?